doc/guidelines/frontend-keyhandling.md

```diff
@@ -0,0 +1,90 @@
+# Frontend Keybindings
+
+This document gives guidelines on how to implement keybindings in the frontend.
+
+### Keybinding Configuration
+
+The file 'keybindings.json' contains all keybindings that are used in the application and serves the following characteristics:
+
+- keybindings can easily be changed by editing the file
+- developers have an overview of all keybindings
+- there are different sections for keybindings:
+  - global: keybindings that are globally active
+  - Scopes for specific layers:
+    Keybindings that are only active if the corresponding layer is active.
+    These scopes should be named according to the layer name, postfixed with `_layer`.
+  - _just_for_documentation_:
+    Keybindings that are only used for documentation purposes since the keybinding is hardcoded in the component.
+    For example, the search field can be cleared by pressing the Escape key, which is typically a native browser action and should not be configurable in the keybindings file.
+- modifier keys are supported (Shift, Alt, Ctrl, Meta)
+  - they can be used in combination with other keys by using the '+' sign (e.g. 'Alt+P', 'Shift+A')
+- multiple keybindings can be assigned to one action
+
+Structure:
+
+```json
+{
+  "global": {
+    "<<action_name>>": ["<<Key-Binding>>", "..."]
+  },
+  "base_layer": {
+    "<<action_name>>": ["<<Key-Binding>>", "..."]
+  },
+  "plants_layer": {
+    "exitPlantingMode": ["Escape"]
+  },
+  "_just_for_documentation_": {
+    "clearSearch": ["Escape"]
+  }
+}
+```
+
+### Keyhandling on Focused Elements
```

Comment:
First describe what should be done in any case. Ideally add a step-by-step description what someone should do to properly add a keybinding. Then add info about special cases.

Reply:
in my opinion the subsections are not describing special cases. there are just two different ways we have to handle keyhandling for either focused elements or keybindings that should be available independent of focus. 
i created a more detailed step-by-step description for both variants.